### PR TITLE
Bug Fix + Cleaned Up Code

### PR DESCRIPTION
**Pull request by varunshenoy**
- Added a new distribution, arcsine, which favors edge values over values closer to the mean
- Bitflow works on the YCbCr case study
- Refactored code and now Bitflow is entirely customizable on each run
- Fully vectorized area and error so the gradient descent achieves the similar values to simulated annealing

### CODE REVIEW

File: BitFlow/Optimization.py [hunk 1]
```diff
@@ -19,11 +19,11 @@ def handleIB(self, node):
         ib = 0
         x = self.node_values[node]
         if isinstance(x, Interval):
-            alpha = 2 if (log2(x.hi) % 1 == 0) else 1
+            alpha = 2 if (log2(abs(x.hi)) % 1 == 0) else 1
```
Comment (rdaly525) [inline BitFlow/Optimization.py]: What is this %1 ? If you are checking if the floating point value is an integer, you can use val.is_integer().

Another (hacky) trick to get the number of bits of an integer is to do len(bin(val))-2

File: output.txt [hunk 2]
```diff
@@ -0,0 +1,149 @@
+
```
Comment (rdaly525) [inline output.txt]: Did you mean to add this file? If you want to start adding results, could you put this into a results folder?

**Review by rdaly525 — CHANGES_REQUESTED** [review]

File: BitFlow/Optimization.py (hunk 1 above)
Comment (varunshenoy) [reply BitFlow/Optimization.py]: I blindly applied the equation in section five of the paper; i'll edit it to use is_integer()

File: output.txt (hunk 2 above)
Comment (varunshenoy) [reply output.txt]: It was just a personal file to compare results — I've removed it.

**Review by rdaly525 — APPROVED** [review]

**Pull request merged**